### PR TITLE
Support OAuth Integrations with probot

### PR DESCRIPTION
**Pull request by pholleran**
## Actions
- [ ] Support calling `bin/probot` with oauth details
- [ ] Support authenticating with github (in `createRobot`) with an oauth token

### CODE REVIEW

**Comment by jbjonesjr** [comment]
Sweet, we're now rebaselined onthe latest probot. woot.

File: lib/robot.js [hunk 1]
```diff
@@ -95,15 +102,18 @@ class Robot {
     return this.events.on(name, async event => {
       if (!action || action === event.payload.action) {
         try {
-          const github = await this.auth(event.payload.installation.id)
+          const github = await this.auth('installation' in event.payload ? event.payload.installation.id : null);
```
Comment (jbjonesjr) [inline lib/robot.js]: this is key because oauth payloads don't include an event.payload.installation object

**Comment by jbjonesjr** [comment]
standard linting problems everywhere...

**Comment by jbjonesjr** [comment]
NOTE: The `block comments` in `lib/robot.js` are all sorts of inconsistent.

**Comment by kenotron** [comment]
ping!

**Comment by gr2m** [comment]
I'm looking into OAuth authentication for GItHub Apps right now. I'm trying to better understand the different use cases that people have, can you share yours @pholleran @kenotron?

**Pull request closed without merging**